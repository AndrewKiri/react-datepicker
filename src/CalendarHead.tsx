import * as React from 'react';

interface Props{
  onPrev?: () => void,
  onNext?: () => void,
  prevIcon?: boolean,
  nextIcon?: boolean,
  title?: string
}

const defaultProps = {
  title: ''
};

const CalendarHead: React.FunctionComponent<Props> = ({
  onPrev,
  onNext,
  prevIcon,
  nextIcon,
  title
}) => {
  return (
    <div className="calendar__head">
      <div className="calendar__head--prev">
        {
          prevIcon && (
            <button 
              onClick={onPrev}
              className="calendar__head--button"
            > 
              <i className="icon-navigate_before"/>
            </button>
          )
        }
      </div>
      <h2 className="calendar__head--title">
        { title }
      </h2>
      <div className="calendar__head--next">
        {
          nextIcon && (
            <button 
              onClick={onNext}
              className="calendar__head--button"
            > 
              <i className="icon-navigate_next"/>
            </button>
          )
        }
      </div>
    </div>
  )
}

CalendarHead.defaultProps = defaultProps;

export default CalendarHead;