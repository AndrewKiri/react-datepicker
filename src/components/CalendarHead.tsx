import * as React from 'react';

interface Props {
  onPrev?: () => void;
  onNext?: () => void;
  onTitleClick?: () => void;
  prevIcon?: boolean;
  nextIcon?: boolean;
  title?: string;
}

const defaultProps = {
  title: '',
};

const CalendarHead: React.FunctionComponent<Props> = ({
  onPrev,
  onNext,
  prevIcon,
  nextIcon,
  title,
  onTitleClick,
}) => {
  return (
    <div className="calendar__head">
      <div className="calendar__head--prev">
        {prevIcon && (
          <button onClick={onPrev} className="calendar__head--button">
            <i className="icon-left" />
          </button>
        )}
      </div>
      <h2 className="calendar__head--title" onClick={onTitleClick}>
        {title}
      </h2>
      <div className="calendar__head--next">
        {nextIcon && (
          <button onClick={onNext} className="calendar__head--button">
            <i className="icon-right" />
          </button>
        )}
      </div>
    </div>
  );
};

CalendarHead.defaultProps = defaultProps;

export default CalendarHead;
