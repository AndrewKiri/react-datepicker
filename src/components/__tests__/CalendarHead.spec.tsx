import { mount, shallow } from 'enzyme';
import * as React from 'react';
import * as sinon from 'sinon';
import CalendarHead from '../CalendarHead';

describe('<CalendarHead/>', () => {
  it('renders CalendarHead with no props', () => {
    const component = shallow(<CalendarHead />);

    expect(component).toMatchSnapshot();
  });

  it('props title correctly', () => {
    const component = shallow(<CalendarHead title="2019/01" />);

    expect(component).toMatchSnapshot();
    expect(component.find('h2.calendar__head--title').text()).toEqual('2019/01');
  });

  it('props prevIcon, nextIcon correctly', () => {
    const component = mount(<CalendarHead prevIcon nextIcon />);

    expect(component).toMatchSnapshot();
    expect(component.find('div.calendar__head--prev > button')).toHaveLength(1);
    expect(component.find('div.calendar__head--next > button')).toHaveLength(1);
  });

  it('props onPrev, onNext correctly', () => {
    const onPrev = sinon.spy();
    const onNext = sinon.spy();

    const component = mount(<CalendarHead prevIcon nextIcon onPrev={onPrev} onNext={onNext} />);

    expect(component).toMatchSnapshot();
    component.find('div.calendar__head--prev > button').simulate('click');
    component.find('div.calendar__head--next > button').simulate('click');

    expect(onPrev).toHaveProperty('callCount', 1);
    expect(onNext).toHaveProperty('callCount', 1);
  });
});
