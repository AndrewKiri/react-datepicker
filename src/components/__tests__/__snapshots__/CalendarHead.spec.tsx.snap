// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarHead/> should props onPrev, onNext correctly 1`] = `
<div
  className="calendar__head"
>
  <div
    className="calendar__head--prev"
  >
    <button
      className="calendar__head--button"
      onClick={[Function]}
    >
      <i
        className="icon-navigate_before"
      />
    </button>
  </div>
  <h2
    className="calendar__head--title"
  />
  <div
    className="calendar__head--next"
  >
    <button
      className="calendar__head--button"
      onClick={[Function]}
    >
      <i
        className="icon-navigate_next"
      />
    </button>
  </div>
</div>
`;

exports[`<CalendarHead/> should props prevIcon, nextIcon correctly 1`] = `
<div
  className="calendar__head"
>
  <div
    className="calendar__head--prev"
  >
    <button
      className="calendar__head--button"
    >
      <i
        className="icon-navigate_before"
      />
    </button>
  </div>
  <h2
    className="calendar__head--title"
  />
  <div
    className="calendar__head--next"
  >
    <button
      className="calendar__head--button"
    >
      <i
        className="icon-navigate_next"
      />
    </button>
  </div>
</div>
`;

exports[`<CalendarHead/> should props title correctly 1`] = `
<div
  className="calendar__head"
>
  <div
    className="calendar__head--prev"
  />
  <h2
    className="calendar__head--title"
  >
    2019/01
  </h2>
  <div
    className="calendar__head--next"
  />
</div>
`;

exports[`<CalendarHead/> should render correctly 1`] = `
<div
  className="calendar__head"
>
  <div
    className="calendar__head--prev"
  />
  <h2
    className="calendar__head--title"
  />
  <div
    className="calendar__head--next"
  />
</div>
`;
